core/src/main/java/org/apache/calcite/interpreter/AggregateNode.java

```diff
@@ -142,6 +143,40 @@ public Accumulator get() {
         return new UdaAccumulatorFactory(
             AggregateFunctionImpl.create(clazz), call, false);
       }
+    } else if (call.getAggregation() == SqlStdOperatorTable.MIN) {
+      switch (call.getType().getSqlTypeName()) {
+      case INTEGER:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MinInt.class), call, true);
+      case BIGINT:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MinLong.class), call, true);
+      case FLOAT:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MinFloat.class), call, true);
+      case DOUBLE:
+      case REAL:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MinDouble.class), call, true);
+      default: throw new UnsupportedOperationException("Unsupported data type for MIN function");
+      }
+    } else if (call.getAggregation() == SqlStdOperatorTable.MAX) {
+      switch (call.getType().getSqlTypeName()) {
+      case INTEGER:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MaxInt.class), call, true);
+      case BIGINT:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MaxLong.class), call, true);
+      case FLOAT:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MaxFloat.class), call, true);
+      case DOUBLE:
+      case REAL:
+        return new UdaAccumulatorFactory(
+            AggregateFunctionImpl.create(MaxDouble.class), call, true);
+      default: throw new UnsupportedOperationException("Unsupported data type for MAX function");
```

Comment:
For consistency, I would prefer using the same pattern as is used for `SUM`. That is, create a `final` variable to hold the class and then make a single call to construct the accumulator. Also, I'm not sure it's necessary to throw an `UnsupportedOperationException` here. (I could be wrong, but I believe types are already checked at this point which is why sum doesn't do this.)

Reply:
Pardon me for the late reply. I completely agree with your point about saving the class type in a variable and initialize the accumulator after the switch.

But I prefer throwing an exception on receiving an unsupported type. This way, a developer won't have to debug through the code if something goes wrong. I know it's highly unlikely. If you insist, I have no problem to comply.

Pardon me for the late response.

Reply:
One concern here is consistency. In any case, I don't see any scenario where this would happen and if it did, I think the exception that would be later thrown would likely be informative enough.

Reply:
Done.